events/views.py

```diff
@@ -543,3 +545,26 @@ def addReply(request, event_id, comment_id):
         return redirect(
             "events:event-detail", event_id=event.id
         )  # redirect to event detail page
+
+
+@login_required
+@require_POST
+def deleteComment(request, comment_id):
+    comment = get_object_or_404(Comment, id=comment_id)
+    # only commenter and event creator can delete a comment/reply
+    if request.user != comment.user and request.user != comment.event.creator:
+        return redirect("events:event-detail", event_id=comment.event.id)
+
+    if request.POST.get("action") == "delete":
+        # only comment with no replies can be deleted
+        if not comment.replies.exists():
+            comment.is_active = False
+            comment.save()
+            return redirect("events:event-detail", event_id=comment.event.id)
+        else:
+            messages.warning(
+                request,
+                "You can't delete this comment. There're replies under this comment.",
```

Comment:
i was wondering if we are validating the error message on frontend as well. Please help me with it.

Reply:
Yes only the comment without replies will have a delete button on frontend